### PR TITLE
DB-5231: Support GraphQL over HTTP GET

### PR DESCRIPTION
**Pull request by mitchellmarkoff**
## What changes were made?
- Support GraphQL over HTTP GET added to the `wordpress-kit`

## Where were the changes made?
- `wordpress-kit`

## How have the changes been tested?
- A GraphQL client was created to support GET requests and the use of this client resulted in successful cache hits.
-   
## Additional information

Don't forget to [add a changeset](https://github.com/pantheon-systems/decoupled-kit-js#generating-a-changeset) if needed!

### CODE REVIEW

**Comment by mitchellmarkoff** [comment]
@CobyPear LMK your thoughts in relation to this work and documentation. I didn't update docs here since `GET` is passed through the options param which is [already documented](https://live-decoupled-kit-docs-canary.appa.pantheon.site/docs/Packages/wordpress-kit/classes/GraphqlClientFactory). Also figured implementation examples would go into the frontend starter docs as opposed to the package docs.

**Comment by CobyPear** [comment]
> Also figured implementation examples would go into the frontend starter docs as opposed to the package docs.

That sounds accurate. Any reason not to do that with this ticket?

**Comment by mitchellmarkoff** [comment]
> That sounds accurate. Any reason not to do that with this ticket?

 I held off on it because of this point in the implementation details thinking that docs updates would come during the starter related work. 
`Changes here are only for wordpress-kit. This will be consumed in the starter kits as part of another ticket.`
With that said, the docs should be simple to write  so if it's easier to add them with this PR that is something I can do.

**Comment by CobyPear** [comment]
> > That sounds accurate. Any reason not to do that with this ticket?
> 
> I held off on it because of this point in the implementation details thinking that docs updates would come during the starter related work. `Changes here are only for wordpress-kit. This will be consumed in the starter kits as part of another ticket.` With that said, the docs should be simple to write so if it's easier to add them with this PR that is something I can do.

oh good call, missed that bit. You can save it for the other ticket. This one should have a test at the very least tho.

File: packages/wordpress-kit/__tests__/GraphqlClientFactory.test.ts [hunk 1]
```diff
@@ -28,3 +28,12 @@ test('request and response middleware add appropriate headers', async () => {
...
 	expect(res.headers.has('surrogate-key-raw')).toBeTruthy();
 });
+
+test('client is configured for GET requests', async () => {
+	const client = new GraphqlClientFactory('http://my-wp.test/graphql').create();
+	const getClient = new GraphqlClientFactory('http://my-wp.test/graphql', {
+		method: 'GET',
+	}).create();
+
+	expect(client).not.toMatchObject(getClient);
```
Comment (CobyPear) [inline packages/wordpress-kit/__tests__/GraphqlClientFactory.test.ts:38]: We can be a bit more specific here:
```suggestion
	const getClient = new GraphqlClientFactory('http://my-wp.test/graphql', {
		method: 'GET',
	});

	expect(getClient.options.method).toEqual('GET');
```

**Review by CobyPear — CHANGES_REQUESTED** [review]

**Review by CobyPear — APPROVED** [review]

**Pull request merged**